kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/PaymentRequestDocument.java

```diff
@@ -348,5 +479,193 @@ public boolean getPayment1099IndicatorForSearching() {
+		} else {
+			value = false;
+		}
+		return value;
+	}
+
+	private void processTaxWithholdingGeneralLedgerPendingEntriesPREQ(GeneralLedgerPendingEntrySequenceHelper sequenceHelper, GeneralLedgerPendingEntrySourceDetail glpeSourceDetail, GeneralLedgerPendingEntry explicitEntry, GeneralLedgerPendingEntry offsetEntry) {
+		String incomeClassCode = ((PaymentRequestDocument) this).getTaxClassificationCode();
+		if ((StringUtils.isNotEmpty(incomeClassCode) && !StringUtils.equalsIgnoreCase(incomeClassCode, "N")) && ((((PaymentRequestDocument) this).getTaxFederalPercent().compareTo(new BigDecimal(0)) != 0) || (((PaymentRequestDocument) this).getTaxStatePercent().compareTo(new BigDecimal(0)) != 0))) {
+			ParameterService parameterService = SpringContext.getBean(ParameterService.class);
+			String taxAccount = parameterService.getParameterValueAsString(PaymentRequestDocument.class, NRATaxParameters.FEDERAL_TAX_PARM_PREFIX + NRATaxParameters.TAX_PARM_ACCOUNT_SUFFIX);
+			if (!offsetEntry.getAccountNumber().equals(taxAccount)) {
+				processTaxWithholdingGeneralLedgerPendingEntries(sequenceHelper, glpeSourceDetail, explicitEntry, offsetEntry, parameterService);
+			}
+		}
+	}
+
+	private void processTaxWithholdingGeneralLedgerPendingEntries(GeneralLedgerPendingEntrySequenceHelper sequenceHelper, GeneralLedgerPendingEntrySourceDetail glpeSourceDetail, GeneralLedgerPendingEntry explicitEntry, GeneralLedgerPendingEntry offsetEntry, ParameterService parameterService) {
+		sequenceHelper.increment();
+
+		GeneralLedgerPendingEntry taxWithholdingExplicit = new GeneralLedgerPendingEntry(explicitEntry);
+		taxWithholdingExplicit.setTransactionLedgerEntrySequenceNumber(sequenceHelper.getSequenceCounter());
+		taxWithholdingExplicit.setTransactionDebitCreditCode(offsetEntry.getTransactionDebitCreditCode());
+		addPendingEntry(taxWithholdingExplicit);
+
+		String glpOffsetObjectCode = parameterService.getParameterValueAsString(KfsParameterConstants.PURCHASING_DOCUMENT.class, PurapParameterConstants.GENERAL_LEDGER_PENDING_ENTRY_OFFSET_OBJECT_CODE);
+		SystemOptions options = SpringContext.getBean(OptionsService.class).getOptions(explicitEntry.getUniversityFiscalYear());
+
+		sequenceHelper.increment();
+
+		GeneralLedgerPendingEntry taxWithholdingOffset = new GeneralLedgerPendingEntry(offsetEntry);
+		taxWithholdingOffset.setTransactionLedgerEntrySequenceNumber(sequenceHelper.getSequenceCounter());
+		taxWithholdingOffset.setFinancialObjectCode(glpOffsetObjectCode);
+		taxWithholdingOffset.setFinancialObjectTypeCode(options.getFinancialObjectTypeAssetsCd());
+		taxWithholdingOffset.setTransactionDebitCreditCode(explicitEntry.getTransactionDebitCreditCode());
+
+		addPendingEntry(taxWithholdingOffset);
+	}
+
+	private void processUseTaxOffsetGeneralLedgerPendingEntries( GeneralLedgerPendingEntrySequenceHelper sequenceHelper, GeneralLedgerPendingEntrySourceDetail glpeSourceDetail, GeneralLedgerPendingEntry explicitEntry, GeneralLedgerPendingEntry offsetEntry) {
```

Comment:
Dead arg: glpeSourceDetail; as a helper, we have control over the method signature, and should remove it .